### ISSUE TITLE
Alt + F4

### ISSUE THREAD

**Issue by AlanGetar** [ppy/osu#8160]
When you use alt+f4 when the game says "See you next time" the music continues to play and osu is open even if it's not opened. And after waiting 30 secs the game turns off.
 
https://www.youtube.com/watch?v=2Bj4ANg5FKw

2020.306.0 

**Logs:**
[performance.log](https://github.com/ppy/osu/files/4300195/performance.log)
[runtime.log](https://github.com/ppy/osu/files/4300196/runtime.log)
[updater.log](https://github.com/ppy/osu/files/4300197/updater.log)
[database.log](https://github.com/ppy/osu/files/4300198/database.log)
[network.log](https://github.com/ppy/osu/files/4300199/network.log)

**Comment by bdach** [comment]
Not sure what this is about. Logs look clean, but I notice the timestamps don't match the time depicted on video - is the video from the same game run as the logs?

The high CPU usage is also curious.

**Comment by AlanGetar** [comment]
possibly my logs are wrong because I got them after shutting osu down.
But you can try it yourself and get logs, this will work because my friends have same problem.

**Comment by bdach** [comment]
Nope, I can't reproduce with any screen mode. Logo fades to black fine, audio fades fine, process exits as soon as the window closes.

**Comment by bdach** [comment]
Doing <kbd>Alt</kbd>+<kbd>F4</kbd> while the "see you next time" sample is playing causes an instant exit here.

**Comment by AlanGetar** [comment]
Tommorow I'll send right logs and more videos, ok?

**Comment by peppy** [comment]
Does the same thing happen when exiting without using alt-f4?
Could you "create dump file" from the right click menu in task manager and send a link to download it (will be quite big). Please note it may contain osu! credentials so I would recommend against sharing it in this thread, or logout before you do so if you do. If you wish to send privately, pe@ppy.sh works.

**Comment by huoyaoyuan** [comment]
> Does the same thing happen when exiting without using alt-f4?

I'm also hitting this by exiting with Esc and select Yes.

Call stack shown when I break it in IDE:
```
Waiting lock hold by thread 12216
... many CompositeDrawable.Dispose()
OsuFocusedOverlayContainer.Dispose()
... many CompositeDrawable.Dispose()
DatabasedKeyBindingContainer<GlobalAction>.Dispose()
... many CompositeDrawable.Dispose()
OsuGameBase.Dispose()
...
```

**Comment by huoyaoyuan** [comment]
Thread 12216 is executing `textures.Get($@"https://a.ppy.sh/{user.Id}");`, and blocked by lock of thread 8204.
Thread 8204 is also acquiring user avatar, and is waiting for `osu.Framework.IO.Network.WebRequest.Perform()`.

**Comment by huoyaoyuan** [comment]
Quick analysis:
`DrawableAvatar` calls `TextureStore.Get()`, which locks a `stores` member and executes synchronized web request.
